server/src/features/facilities/facility.controller.js

```diff
@@ -20,6 +25,26 @@ export class FacilitiesController {
     }
   }
 
+  async reserveCourt(req, res, next) {
+    try {
+      if (!req.user) {
+        throw new ApiError(401, "Unauthorized");
+      }
+      const reservation = await FacilitiesService.reserveCourt(
+        req.user.id,
+        req.body
+      );
+
+      res.status(201).json({
+        success: true,
+        message: "Court reserved successfully.",
+        data: reservation,
+      });
+    } catch (error) {
```

Comment:
```suggestion
   } catch (err) {
  if (err.code === 11000) {
    return res.status(409).json({
      message: "This time slot is already booked"
    });
  }

  next(err);
}

```
for duplicate bookings